service/controller/v5/cloudconfig/cloud_config.go

```diff
@@ -115,15 +123,14 @@ func (c CloudConfig) NewMasterCloudConfig(customObject providerv1alpha1.AzureCon
 	customObject.Spec.Cluster.Calico.Subnet = c.azureNetwork.Calico.IP.String()
 	customObject.Spec.Cluster.Calico.CIDR, _ = c.azureNetwork.Calico.Mask.Size()
 
-	params := k8scloudconfig.Params{
```

Comment:
Why the initialization changed here?

We usually do something like below. Initializing fields in between struct fields settings make things hard to read IMO.

```go
	ignitionPath := k8scloudconfig.GetIgnitionPath(c.ignitionPath)
	files, err := k8scloudconfig.RenderFiles(ignitionPath, params)
	if err != nil {
		return "", microerror.Mask(err)
	}

	params = k8scloudconfig.Params{
		...
	}

```

Reply:
I think it changes because of this https://github.com/giantswarm/k8scloudconfig/blob/master/v_4_1_0/cloudconfig.go#L32-L41 There are some default parameters already, which we need on the files rendering stage and not on the cloudconfig init  https://github.com/giantswarm/k8scloudconfig/blob/master/v_4_1_0/cloudconfig.go#L49

Reply:
Why do we have those `DefaultParams`? Why aren't they set in `NewCloudConfig`? That should be the place for that.

Reply:
They are not there because NewCloudConfig is executed after rendering files. So if there are templates, which needs some of those parameters - those templates are generated with empty values (e.g. empty etcd port)

Reply:
Ack. Then we should do defaulting after setting params. Also if we do rendering before calling NewCloudConfig why does it even need to know about those params? But let's not change this in this PR.

Reply:
How can we do defaulting after settings params if setting params overrides defaulting?

Reply:
You don't set the default if the value is already set. E.g. if EtcdPort is set to 1234 you don't change it. If it is 0 then you set it to 443.

Reply:
Ok, I'll add the issue for this